### PR TITLE
Fixes navigation back to main menu in demo game

### PR DESCRIPTION
**Pull request by vlidholt**

### CODE REVIEW

**Comment by Hixie** [comment]
Why does the Navigator.of() convention not work here?

File: examples/game/lib/main.dart [hunk 1]
```diff
@@ -96,6 +96,8 @@ class GameDemoState extends State<GameDemo> {
   }
 
   Widget _buildMainScene(RouteArguments args) {
+    NavigatorState navigatorState = Navigator.of(args.context);
```
Comment (abarth) [inline examples/game/lib/main.dart]: navigatorState <--- I'd just call this navigator.

**Comment by abarth** [comment]
> Why does the Navigator.of() convention not work here?

The problem is that the second callback is called when the game is over.  At that point, we're no longer showing the "welcome" route.  Instead, we're showing the game route, which means the context of the welcome route isn't in the tree anymore and therefore can't find the navigator.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
@vlidholt Did you want to merge this patch?

**Pull request merged**